src/annotator/test/frame-observer-test.js

```diff
@@ -51,71 +74,48 @@ describe('annotator/frame-observer', () => {
     });
 
     it('triggers onFrameAdded when an annotatable iframe is added', async () => {
-      let iframe = createAnnotatableIFrame();
-      await delay(0);
-
-      assert.calledWith(onFrameAdded, iframe);
-
-      iframe = createAnnotatableIFrame('enable-annotation', 'yes');
-      await delay(0);
-
-      assert.calledWith(onFrameAdded, iframe);
-
-      iframe = createAnnotatableIFrame('enable-annotation', 'true');
-      await delay(0);
-
-      assert.calledWith(onFrameAdded, iframe);
-
-      iframe = createAnnotatableIFrame('enable-annotation', '1');
-      await delay(0);
-
-      assert.calledWith(onFrameAdded, iframe);
-
-      iframe = createAnnotatableIFrame('enable-annotation', 'false'); // the actual value of the attribute is irrelevant
-      await delay(0);
-
+      const iframe = createAnnotatableIFrame();
+      await waitForCall(onFrameAdded);
       assert.calledWith(onFrameAdded, iframe);
     });
 
     it("doesn't trigger onFrameAdded when non-annotatable iframes are added", async () => {
       createAnnotatableIFrame('dummy-attribute');
-      await delay(0);
-
+      await delay(10);
```

Comment:
This test asserts a callback was _not_ invoked, so we need to use a delay, whereas the other tests can wait for the spy to be called. The value is arbitrary, but it needs to be ideally the lowest value such that the callback would very likely be invoked if it was going to be invoked.